add Box.com as oauth2 provider. Details :

blakepatteson opened the pull request:
1.  implement Box OAuth2 authentication provider, following pattern of all the others
2.  add Box provider configuration to UI
3.  include Box logo .svg file and tests
4.  update provider count in tests

I built / tested the front-end after running `npm run build` but deleted the js bundles so as to keep the diff clean... I'll happily rebuild it if requested.

  API reference links :
      https://developer.box.com/reference/get-users-me/
      https://developer.box.com/guides/authentication/oauth2/oauth2-setup/

blakepatteson committed:
add Box.com as oauth2 provider. Details :

1.  implement Box OAuth2 authentication provider, following pattern of
    all the others
2.  add Box provider configuration to UI
3.  include Box logo .svg file and tests
4.  update provider count in tests

  API reference links :
      https://developer.box.com/reference/get-users-me/
      https://developer.box.com/guides/authentication/oauth2/oauth2-setup/

ganigeorgiev commented on tools/auth/box.go:
```diff
@@ -0,0 +1,101 @@
+		Id:           extracted.ID,
+		Name:         extracted.Name,
+		Username:     extracted.Login,
+		Email:        extracted.Login, // Box uses login as email identifier
```
Please double check that the `extracted.Login` field is a valid email address and that it is always verified.

ganigeorgiev commented on tools/auth/box.go:
```diff
@@ -0,0 +1,101 @@
+	user := &AuthUser{
+		Id:           extracted.ID,
+		Name:         extracted.Name,
+		Username:     extracted.Login,
```
If the provider doesn't return a username, do not assign it to the email and leave it empty instead so that it can be autogenerated.

ganigeorgiev reviewed (changes requested):
Thank you, I'll have a more detailed look at it eventually this weekend _(or the next one)_.

For now I left some nitpicks that you could check when you find time.

blakepatteson commented:
Thanks so much for the feedback! 

I will make these updates shortly and mark the pull request as "Ready for Review". 

PS : thank you for putting PocketBase out there - it's an amazing tool - so simple, but still useful in our times of increasingly complicated software. and thanks for making it easy to contribute!

blakepatteson committed:
address all feedback points from review. Details :

1.  rm unused fields from extracted struct from box api user response
2.  add check to ensure box accounts are active to prevent auth with
    disabled Box accounts
3.  confirmed with manual testing that Box enforces email verification
    at the OAuth authorization level - unverified users are prompted to
    "please click the confirmation link" for the Oauth flow to even work
4.  rm username assignment so pocketbase can autogenerate it
5.  change scope from root_readwrite to root_readonly following
    principle of least privilege
